plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -198,6 +193,14 @@ def static_env
       kube_cluster_name = DeployGroup.find(metadata[:deploy_group_id]).kubernetes_cluster.name.to_s
       env[:KUBERNETES_CLUSTER_NAME] = kube_cluster_name
 
+      # the vault server to be used if we are using vault and there's an instance associated
+      # with the deploy group
+      if @doc.deploy_group.vault_server_id
```

Comment:
might be better to fail fast here and not after the deploy is underway

-> require there to be a vault server if secrets are used and vault backend is used ?